### ISSUE TITLE
I put in a bad host and now it's stuck

### ISSUE THREAD

**Issue by danielb2** [bobthecow/genghis#44]
I checked `~/.vegas/genghisapp`  for any clues, but restarting the server hasn't helped. It's just spinning waiting for the hosts to load.

I'm not sure what was bad about my connection but it got stuck after a new one. How can I reset genghis to default?

**Comment by bobthecow** [comment]
Clear the Genghis cookies. 

Can you tell me what sort of invalid hostname is failing? It should handle those pretty okay.

**Comment by danielb2** [comment]
Looks like it was an edge case. Took me a few tries to pin it down.

`mongodb://user:pass@ds037977.mongolab.com:37977/some_app`

btw, I submitted proper credentials. Maybe creating an account with them will reveal the issue

**Comment by danielb2** [comment]
( I cannot reproduce with any other host )

**Comment by bobthecow** [comment]
Sorry about that. Adding to the pile of server improvement tasks in the v2.2 queue: #16, #37 and #42

**Comment by tonyxiao** [comment]
same issue. trying to use genghisapp with mongohq url. `mongodb://heroku:adkwlelwwe342u32l@alex.mongohq.com:10221/app23232` But genghisapp keeps getting stuck. Running in foreground shows this error. 

```

Mongo::OperationFailure - Database command 'listDatabases' failed: (errmsg: 'need to login'; ok: '0.0').:
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/mongo-1.7.0/lib/mongo/db.rb:535:in `command'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/genghisapp-2.1.3/genghis.rb:480:in `info'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/genghisapp-2.1.3/genghis.rb:459:in `as_json'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/genghisapp-2.1.3/genghis.rb:470:in `to_json'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-contrib-1.3.1/lib/sinatra/json.rb:119:in `to_json'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-contrib-1.3.1/lib/sinatra/json.rb:119:in `json'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/genghisapp-2.1.3/genghis.rb:696:in `block in <class:Server>'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:1265:in `call'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:1265:in `block in compile!'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:835:in `[]'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:835:in `block (3 levels) in route!'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:851:in `route_eval'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:835:in `block (2 levels) in route!'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:872:in `block in process_route'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:870:in `catch'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:870:in `process_route'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:834:in `block in route!'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:833:in `each'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:833:in `route!'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:936:in `dispatch!'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:769:in `block in call!'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:921:in `block in invoke'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:921:in `catch'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:921:in `invoke'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:769:in `call!'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:755:in `call'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/rack-protection-1.2.0/lib/rack/protection/xss_header.rb:22:in `call'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/rack-protection-1.2.0/lib/rack/protection/path_traversal.rb:16:in `call'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/rack-protection-1.2.0/lib/rack/protection/json_csrf.rb:17:in `call'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/rack-protection-1.2.0/lib/rack/protection/base.rb:47:in `call'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/rack-protection-1.2.0/lib/rack/protection/xss_header.rb:22:in `call'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/rack-1.4.1/lib/rack/nulllogger.rb:9:in `call'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/rack-1.4.1/lib/rack/head.rb:9:in `call'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:99:in `call'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:1389:in `block in call'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:1471:in `synchronize'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/sinatra-1.3.3/lib/sinatra/base.rb:1389:in `call'
    /Users/Tony/.rvm/gems/ruby-1.9.3-p194/gems/rack-1.4.1/lib/rack/handler/webrick.rb:59:in `service'
    /Users/Tony/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/1.9.1/webrick/httpserver.rb:138:in `service'
    /Users/Tony/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/1.9.1/webrick/httpserver.rb:94:in `run'
    /Users/Tony/.rvm/rubies/ruby-1.9.3-p194/lib/ruby/1.9.1/webrick/server.rb:191:in `block in start_thread'
localhost - - [07/Nov/2012:15:13:25 PST] "GET /servers HTTP/1.1" 500 103
http://0.0.0.0:5678/ -> /servers
```

**Comment by bobthecow** [comment]
This is fixed in `develop` with saner connection timeouts, and will be in the next release.

@tonyxiao Your issue is slightly different, and will be fixed with #16
